### ISSUE TITLE
update: xbar doesn't restart after update

### ISSUE THREAD

**Issue by matryer** [matryer/xbar#631]
While the in-built restart feature does update the app, when it tries to start the new version and self-terminate, the new version never seems to start.

@leaanthony Did you mention that you had an idea about this?

**Comment by leaanthony** [comment]
No but I am seeing this behaviour in more recent builds. Pretty certain this used to work

**Comment by matryer** [comment]
Right now it does this:

1. Download latest release alongside xbar.app
2. Unpack the tar to overwrite the existing xbar.app
3. Start the new app
4. Wait two seconds and terminate self

I don't know if macOS is preventing multiple versions os xbar from running or something.

Maybe it's better for the code to write a script that will replace the binary and relaunch the app after a delay. Then xbar can quit safely, and the script can take over?

**Comment by ggrelet** [comment]
I don't know if that's related but the update popup does not have its icon right: 
<img width="488" alt="Capture d’écran 2021-03-15 à 14 02 01" src="https://user-images.githubusercontent.com/10114131/111157435-2c04cc80-8597-11eb-9136-8b693382b313.png">

**Comment by matryer** [comment]
@leaanthony can you put the logos in so I can edit them pleeeease?

**Comment by leaanthony** [comment]
Copied, pushed and briefly tested:
![image](https://user-images.githubusercontent.com/1943904/111769465-5e972800-88fd-11eb-9835-0627738f3774.png)

**Comment by leaanthony** [comment]
Suggestion: Why not put the update notification in the menu? If we want to use "Start at Login", having a dialog flash up is probably not what we want in terms of UX. If it's not obvious enough, we could have a new tray icon indicating an update is available? Something like this: 
![image](https://user-images.githubusercontent.com/1943904/111857528-40c1d580-8986-11eb-84c2-1b62954825b4.png)

**Comment by magicus** [comment]
Putting "xbar update available!" in the menu bar would also not be optimal UX :-). But I think that you could change the current "Check for update" to a "Update to xbar v2.0.32", if an update is available. That it, it seems reasonable that the app performs a check for available updates in the background, and makes it clear (but not obnoxiously so) that an update is available.

And possibly that you indicate this fact by adding this text to the main menu. But I'd personally prefer that you just signalled that an update was available, so something like the last screenshot above, but the text being "xbar x2.0.32 available" and in smaller, greyish font, like how the current xbar version is displayed in the xbar menu right now.

**Comment by matryer** [comment]
Yeah I like this. Maybe an auto-update option too, so it just sorts itself
out in the background.

The dialog is annoying.

**Comment by magicus** [comment]
As long as you can disable the auto update... I hate it when stuff change under your feet, potentially breaking things.
